File: Implementierung/Backend/src/main/java/edu/kit/informatik/dto/UserDto.java
package edu.kit.informatik.dto;

import edu.kit.informatik.model.User;


import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.sql.Timestamp;
import java.util.UUID;

/**
 * Data-Transfer-Object zur Klasse {@link User}
 *
 * @author uekai
 * @author ugqbo
 * @version 1.0
 */
@ToString
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class UserDto implements Comparable<UserDto> {

    private String id;
    private String firstName;
    private String lastName;
    private String email;
    private String password;
    private boolean auth;
    private String token;
    private RoleDto role;
    private Timestamp createdAt;
    private Timestamp updatedAt;

    @Override
    public int compareTo(UserDto o) {
        UUID thisUser = UUID.fromString(this.id);
        UUID oUser = UUID.fromString(o.getId());

        return  thisUser.compareTo(oUser);
    }
}

File: Implementierung/frontend/paschi/src/dto/UserDto.ts
import {RoleDto} from "@/dto/RoleDto";

export class UserDto {

  id: string;
  createdAt: string;
  updatedAt: string;
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  auth: boolean ;
  role: RoleDto;
  token: string | undefined;

  constructor(id: string, createdAt: string, updatedAt: string, firstName: string, lastName: string, email: string,
              password: string, auth: boolean, role: RoleDto, token: string | undefined) {
    this.id = id;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.firstName = firstName;
    this.lastName = lastName;
    this.email = email;
    this.password = password;
    this.auth = auth;
    this.role = role;
    this.token = token;
  }
}
